bootable-jar/WFLY-14934_Server_Mode.adoc

```diff
@@ -0,0 +1,123 @@
+
+* Plugin behave the same as when producing a Bootable JAR, the only visible difference is that the project target dir contains 
+   a directory (named server by default) that contains the server and any deployments.
+* The Server mode is activated by the ```<server>``` configuration element. 
+* The server mode can be enabled/disabled, e.g.: ```<server> <enabled>[true|false]</enabled></server>```
+* In server mode, multiple deployments can be deployed inside the server. The primary one (the current project if producing a primary artifact) 
+   but also any war/ear files added as Maven dependencies in the project pom.xml file.
+* In bootable JAR mode, a single deployment can be added (the primary or dependency one).
+* By default the server is generated in ```target/server``` directory. The plugin offers a configuration item to override the name of the directory generated in the
+   ```target``` directory. E.g.: ```<server> <directory-name>foo-server</directory-name></server>```
+* In server mode, in order to allow for execution of a launch script for start/run/dev/dev-watch modes in a cloud 
+   context (eg: ODO), a launcher bash script file name (located in $JBOSS_HOME/bin) can be provided. For example, 
+   when using the legacy cloud galleon feature-pack, one can provide the ``` openshift-launch.sh``` script file name that this galleon feature-pack provisions.
+   For example: ```<server> <launch-script>openshift-launch.sh</launch-script></server>```
+* ```<cloud>``` configuration applies to the server mode the same way it applies to Bootable JAR. The Bootable JAR specific cloud extension is not packaged in the server.
+It is expected that the cloud execution context (e.g.: the WildFly S2i v2 builder and runtime images 
+defined in https://issues.redhat.com/browse/WFLY-14936[WFLY-14934]) handles the specifics required for the WildFly server to behave properly.
+* ```dev, dev-watch, start, run and shutdown``` goals work the same way as when Bootable JAR packaging occurs.
+
+==== Requirements for cloud integration
+
+* The plugin configured in this server mode can be used in a "cloud build" context. In particular, the plugin must be usable with:
+** WildFly s2i V2 builder image for s2i in binary and source (see: https://issues.redhat.com/browse/WFLY-14936[WFLY-14934])
+** JKube s2i and docker binary builds (see https://issues.redhat.com/browse/WFLY-14936[WFLY-14938])
+** Other technologies such as Buildpacks.
+* The generated server can be installed and run successfully inside a WildFly runtime image v2 defined 
+   in https://issues.redhat.com/browse/WFLY-14936[WFLY-14934].
+
+==== Impact on quickstarts 
+
+* A set of quickstarts (List TBD) will be refactored to use this server mode in both bare-metal and cloud.
+
+=== Nice-to-Have Requirements
+
+* NONE
+
+=== Non-Requirements
+
+* No support for domain mode.
+* The ability to execute a launch bash script file is not multi platforms. \WildFly cloud images is the only supported platform.
```

Comment:
I don't understand what that means.

Reply:
I clarified, we shouldn't support anything else than ODO use-case: the Bootable JAR Maven plugin is executed in the cloud in a docker image. For example, Windows platform or Mac would be not supported.